airflow/serialization/serialized_objects.py

```diff
@@ -195,6 +196,19 @@ def serialize_to_json(
                 serialized_object[key] = value
         return serialized_object
 
+    @classmethod
+    def serialize_timezone(cls, var: datetime.tzinfo) -> Any:
+        """Serializes a timezone to json"""
+        tz_name = getattr(var, 'name', None)
+        if not tz_name:
```

Comment:
Can we also use `tzname` for pendulum Timezone as well? It’s a standard method that should always yield something useful.

Reply:
> It's a standard method that should always yield something useful

Well, not always with pendulum as explained in https://github.com/apache/airflow/pull/16631#issuecomment-867612620 , `pendulum.timezone('Europe/Stockholm').tzname(...)` can return `CEST` which **can't** be converted back with `pendulum.timezone('CEST')` as it raises InvalidTimezone

I did try that at some point. The whole concept of `datetime.tzinfo` -> `str` -> `datetime.tzinfo` works sometimes but there is no really hard guarantee. So that's why I use the `.name` if present (which marginally increases the chances of success serialization/deserialization)

Reply:
Gosh we really should drop Pendulum at some point I guess.

Reply:
From what I can tell there is _no_ good timezone library in python really :(

Reply:
There are no good timezone libraries, in general 🙂 Personally I think the new `zoneinfo` is as good as it can be.